aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -215,7 +215,44 @@ Tensor& cumprod_out(Tensor& result, const Tensor& self, int64_t dim, c10::option
   return result;
 }
 
+std::tuple<Tensor&, Tensor&> cummax_out(Tensor& out, Tensor& indices, const Tensor& self, int64_t dim, c10::optional<ScalarType> dtype) {
+  TORCH_CHECK(
+      !dtype.has_value() || (out.scalar_type() == dtype.value()),
+      "provided dtype must match dtype of result in cummax. Got ",
+      toString(out.scalar_type()),
+      " and ",
+      toString(dtype.value()),
+      ".");
+  if(self.dim() == 0) {
+    out = self;
+    indices = self;
+  }
+  else {
+    NoNamesGuard guard;
+    //update out_ and indices_ for the first values along the dimension dim
+    auto out_ = self.narrow(dim, 0, 1);
+    auto indices_ = at::zeros(self.narrow(dim, 0, 1).sizes());
+    for(int64_t i=1; i<self.size(dim); i++) {
+      auto res_at_i = at::native::max(at::cat({out_.narrow(dim, i-1, 1), self.narrow(dim, i, 1)}, dim), dim);
+      auto out_at_i = (std::get<0>(res_at_i)).unsqueeze(dim);
+      out_ = at::cat({out_, out_at_i}, dim);
+      auto max_indices_at_i = at::max(indices_.narrow(dim, i-1, 1), (i * (std::get<1>(res_at_i)).unsqueeze(dim)).toType(at::kDouble));
+      indices_ = at::cat({indices_, max_indices_at_i}, dim);
+    }
+    out=out_;
+    indices=indices_;
+  }
+  namedinference::propagate_names(out, self);
+  namedinference::propagate_names(indices, self);
+  return std::tuple<Tensor &,Tensor &>{out, indices};
+}
 
+std::tuple<Tensor, Tensor> cummax(const Tensor& self, int64_t dim, c10::optional<ScalarType> dtype) {
+  auto out = at::empty(self.sizes(), self.options());
```

Comment:
Shouldn't `out` be constructed using the given `dtype`? Otherwise, it is just ignored no?

Reply:
thanks for the catch! fixed it